File: test/three/app/src/main/java/com/z3d/frame3d/FrameLinePointVo.java
package com.z3d.frame3d;

import org.json.JSONObject;

import com.z3d.base.Object3D;

public class FrameLinePointVo extends Object3D {
    public float time;
    public int id;
    public boolean iskeyFrame;
    public boolean isAnimation;
    public Object data;


    public void writeObject(JSONObject $obj) {

        try {

            this.time = $obj.getInt("time");
//            this.id = $obj.get("id")!=null?(int) $obj.get("id"):0;
            this.iskeyFrame = $obj.getBoolean("iskeyFrame");
            this.isAnimation = $obj.getBoolean("isAnimation");


            this.x =  (float) $obj.getDouble("x") / 10;
            this.y =  (float)  $obj.getDouble("y")  / 10;
            this.z =  (float) $obj.getDouble("z")  / 10;

            this.scaleX = (float) $obj.getDouble("scaleX") / 10;
            this.scaleY =  (float)  $obj.getDouble("scaleY")  / 10;
            this.scaleZ =  (float)  $obj.getDouble("scaleZ")  / 10;

            this.rotationX =   (float) $obj.getDouble("rotationX") ;
            this.rotationY =  (float) $obj.getDouble("rotationY") ;
            this.rotationZ =  (float) $obj.getDouble("rotationZ") ;

            this.data = $obj.get("data");


           

        } catch (Exception e) {
            e.printStackTrace();
        }




    }
}

File: metalios/listios/StorageStarted/pan3d/frame3d/FrameLinePointVo.m
//
//  FrameLinePointVo.m
//  iosgl
//
//  Created by pan3dme on 2020/11/16.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "FrameLinePointVo.h"

@implementation FrameLinePointVo
-(void)writeObject:(NSDictionary*)val ;
{
    FrameLinePointVo* this=self;
    
    this.time = [[val valueForKey:@"time"]intValue];
    
    this.iskeyFrame = [[val valueForKey:@"iskeyFrame"]boolValue];
    this.isAnimation =[[val valueForKey:@"isAnimation"]boolValue];
    
    this.x =  [[val valueForKey:@"x"]floatValue]/ 10;
    this.y =  [[val valueForKey:@"y"]floatValue]  / 10;
    this.z =   [[val valueForKey:@"z"]floatValue]  / 10;
    
    this.scaleX =  [[val valueForKey:@"scaleX"]floatValue] / 10;
    this.scaleY =  [[val valueForKey:@"scaleY"]floatValue]/ 10;
    this.scaleZ =   [[val valueForKey:@"scaleZ"]floatValue] / 10;
    
    this.rotationX =  [[val valueForKey:@"rotationX"]floatValue];
    this.rotationY =  [[val valueForKey:@"rotationY"]floatValue] ;
    this.rotationZ =   [[val valueForKey:@"rotationZ"]floatValue];
    this.data = [val valueForKey: @"data"];
 

}
@end
